app/models/program.rb

```diff
@@ -49,7 +51,7 @@ def confirmed
 #   validates :conference_id, presence: true, uniqueness: true
   validates :rating, numericality: { greater_than_or_equal_to: 0, less_than_or_equal_to: 10 }
 
-  before_create :create_event_types
+  after_create :create_event_types
```

Comment:
Because of versioning, on trying to create any factory that needs program, it'll throw up an error saying the factory is invalid.
The reason is that on creating a program this before_create creates  2 event_types without a program id.This triggers paper_trail to record the create event of these EventTypes  resulting in a call to the EventType method conference_id which throws up `undefined method conference_id for nil:NilClass (NoMethodError)` as these 2 event types do not have a program at that point.
I included these fixes as the errors are  direct consequences of the changes introduced in this PR.